### ISSUE TITLE
Rewrite with NodeJS and Typescript

### ISSUE THREAD

**Issue by polletfa** [polletfa/rmWebUI#30]
Rewrite with NodeJS and Typescript because I strongly dislike PHP as it turns out... :-/

- Stand-alone application (not relying on a separate WebServer)
- NodeJS web server serving:
  - the frontend as a single file webapplication + favicon
    - build script to generate the frontend (using special include statements)
  - APIs:
    - /cloud/* -> Access the cloud - Dummy implementationfor this issue - separate issues for the actual implementations: #31 and #32
    - /backend/* -> Internal functionalities of the backend (e.g. login, logout...)
- HTTP session management required to implement the demo mode
- Multiple servers in parallel possible
- YAML config file
- Backend includes some configuration information as a JSON file inside the frontend file (using a marker)
- Tools/Build/Deployment in Typescript/NodeJS as well

Note: the implementation of the cloud API is out of scope of this issue
